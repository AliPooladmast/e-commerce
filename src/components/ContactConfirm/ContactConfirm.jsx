import style from "./contactConfirm.module.scss";

const ContactConfirm = ({ select, setSelect, setInput, currentUser }) => {
  const handleSelect = (e) => {
    setSelect((prev) => ({ ...prev, [e.target.name]: e.target.id }));
  };

  const handleInput = (e) => {
    setInput((prev) => ({ ...prev, [e.target.name]: e.target.value }));
  };

  return (
    <div className={style.UserInfo}>
      <div className={style.InfoContainer}>
        <div className={style.Option}>
          <input
            name="phone"
            type="radio"
            id="defaultPhone"
            onChange={handleSelect}
            defaultChecked
          />
          <label htmlFor="defaultPhone">Default phone:</label>
          <span>{currentUser?.phone}</span>
        </div>

        <div className={style.Option}>
          <input
            name="phone"
            type="radio"
            id="newPhone"
            onChange={handleSelect}
          />
          <label htmlFor="newPhone">New phone:</label>
          <input
            name="phone"
            placeholder="new phone"
            className={style.Phone}
            onChange={handleInput}
            disabled={select.phone === "defaultPhone"}
          />
        </div>
      </div>

      <div className={style.InfoContainer}>
        <div className={style.Option}>
          <input
            name="address"
            type="radio"
            id="defaultAddress"
            defaultChecked
            onChange={handleSelect}
          />
          <label htmlFor="defaultAddress">Default address:</label>
          <span>{currentUser?.address}</span>
        </div>

        <div className={style.Option}>
          <input
            name="address"
            type="radio"
            id="newAddress"
            onChange={handleSelect}
          />
          <label htmlFor="newAddress">New address:</label>
          <div>
            <textarea
              name="address"
              placeholder="new address"
              className={style.AddressText}
              onChange={handleInput}
              disabled={select.address === "defaultAddress"}
            />
          </div>
        </div>
      </div>
    </div>
  );
};

export default ContactConfirm;
