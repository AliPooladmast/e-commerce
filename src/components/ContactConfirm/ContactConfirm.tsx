import style from "./contactConfirm.module.scss";

interface IContact {
  phone: string;
  address: string;
}
interface ITarget {
  name: string;
  id?: string;
  value?: string;
}

const ContactConfirm = ({
  select,
  setSelect,
  setInput,
  currentUser,
}: {
  select: IContact;
  setSelect: (prev: object) => void;
  setInput: (prev: object) => void;
  currentUser: IContact;
}) => {
  const handleSelect = (e: { target: ITarget }) => {
    setSelect((prev: object) => ({ ...prev, [e.target.name]: e.target.id }));
  };

  const handleInput = (e: { target: ITarget }) => {
    setInput((prev: object) => ({ ...prev, [e.target.name]: e.target.value }));
  };

  return (
    <div className={style.UserInfo}>
      <div className={style.InfoContainer}>
        <div className={style.Option}>
          <input
            name="phone"
            type="radio"
            id="defaultPhone"
            onChange={handleSelect}
            defaultChecked
          />
          <label htmlFor="defaultPhone">Default phone:</label>
          <span>{currentUser?.phone}</span>
        </div>

        <div className={style.Option}>
          <input
            name="phone"
            type="radio"
            id="newPhone"
            onChange={handleSelect}
          />
          <label htmlFor="newPhone">New phone:</label>
          <input
            name="phone"
            placeholder="new phone"
            className={style.Phone}
            onChange={handleInput}
            disabled={select.phone === "defaultPhone"}
          />
        </div>
      </div>

      <div className={style.InfoContainer}>
        <div className={style.Option}>
          <input
            name="address"
            type="radio"
            id="defaultAddress"
            defaultChecked
            onChange={handleSelect}
          />
          <label htmlFor="defaultAddress">Default address:</label>
          <span>{currentUser?.address}</span>
        </div>

        <div className={style.Option}>
          <input
            name="address"
            type="radio"
            id="newAddress"
            onChange={handleSelect}
          />
          <label htmlFor="newAddress">New address:</label>
          <div>
            <textarea
              name="address"
              placeholder="new address"
              className={style.AddressText}
              onChange={handleInput}
              disabled={select.address === "defaultAddress"}
            />
          </div>
        </div>
      </div>
    </div>
  );
};

export default ContactConfirm;
