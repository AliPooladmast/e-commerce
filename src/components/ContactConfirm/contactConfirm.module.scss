.UserInfo {
  flex: 1;
}

.InfoContainer {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin: 10px 0 20px 30px;
  border: 1px solid lightgrey;
  width: fit-content;
  padding: 20px 20px 10px 20px;
  border-radius: 5px;

  .Option {
    margin-bottom: 10px;

    > span {
      color: darkblue;
      font-size: 14px;
    }

    > input {
      margin-right: 5px;
      cursor: pointer;
    }

    > label {
      margin-right: 5px;
      font-weight: 600;
      cursor: pointer;
    }

    .Phone {
      border: 1px solid lightgrey;
      border-radius: 5px;
      padding: 5px 10px;
      font-size: 14px;
      cursor: auto;

      &:focus {
        outline: none;
      }
    }
  }

  .AddressText {
    width: 350px;
    resize: none;
    height: 60px;
    border: 1px solid lightgrey;
    border-radius: 5px;
    padding: 10px;
    font-size: 14px;
    margin-top: 5px;

    &:focus {
      outline: none;
    }
  }
}
