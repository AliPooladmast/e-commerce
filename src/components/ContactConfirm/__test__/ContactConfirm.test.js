const { render, screen } = require("@testing-library/react");
const { BrowserRouter } = require("react-router-dom");
import ContactConfirm from "../ContactConfirm";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";

describe("check the inputs", () => {
  const currentUser = { phone: "123", address: "aaa" };
  const select = {
    phone: "defaultPhone",
    address: "defaultAddress",
  };

  beforeEach(() => {
    render(
      <BrowserRouter>
        <ContactConfirm currentUser={currentUser} select={select} />
      </BrowserRouter>
    );
  });

  it("check if default cases are selected", () => {
    expect(screen.getByTestId("defaultPhone")).toBeChecked();
    expect(screen.getByTestId("defaultAddress")).toBeChecked();
  });

  it("check if new cases are selected", () => {
    expect(screen.getByTestId("newPhone")).not.toBeChecked();
    expect(screen.getByTestId("newAddress")).not.toBeChecked();
  });

  it("check if phone of current user is shown correctly", () => {
    expect(screen.getByTestId("user-phone").textContent).toBe(
      currentUser.phone
    );
  });

  it("check if address of current user is shown correctly", () => {
    expect(screen.getByTestId("user-address").textContent).toBe(
      currentUser.address
    );
  });
});
