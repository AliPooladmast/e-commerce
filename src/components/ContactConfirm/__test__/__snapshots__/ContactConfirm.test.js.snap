// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot match the snapshot 1`] = `
<div
  className="UserInfo"
>
  <div
    className="InfoContainer"
  >
    <div
      className="Option"
    >
      <input
        data-testid="defaultPhone"
        defaultChecked={true}
        id="defaultPhone"
        name="phone"
        onChange={[Function]}
        type="radio"
      />
      <label
        htmlFor="defaultPhone"
      >
        Default phone:
      </label>
      <span
        data-testid="user-phone"
      >
        123
      </span>
    </div>
    <div
      className="Option"
    >
      <input
        data-testid="newPhone"
        id="newPhone"
        name="phone"
        onChange={[Function]}
        type="radio"
      />
      <label
        htmlFor="newPhone"
      >
        New phone:
      </label>
      <input
        className="Phone"
        disabled={true}
        name="phone"
        onChange={[Function]}
        placeholder="new phone"
      />
    </div>
  </div>
  <div
    className="InfoContainer"
  >
    <div
      className="Option"
    >
      <input
        data-testid="defaultAddress"
        defaultChecked={true}
        id="defaultAddress"
        name="address"
        onChange={[Function]}
        type="radio"
      />
      <label
        htmlFor="defaultAddress"
      >
        Default address:
      </label>
      <span
        data-testid="user-address"
      >
        aaa
      </span>
    </div>
    <div
      className="Option"
    >
      <input
        data-testid="newAddress"
        id="newAddress"
        name="address"
        onChange={[Function]}
        type="radio"
      />
      <label
        htmlFor="newAddress"
      >
        New address:
      </label>
      <div>
        <textarea
          className="AddressText"
          disabled={true}
          name="address"
          onChange={[Function]}
          placeholder="new address"
        />
      </div>
    </div>
  </div>
</div>
`;
